### PR TITLE
Feature/127 section organization

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [EQ-127](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-127)
* [EQ-142](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-142)

## Main Changes:
* Added accordion sections to the user interface.
* Re-positioned the "Clear search" button directly beneath the filter inputs.
* Added range inputs to the UI for columns with year and date values.
* Added range handling logic to the server.
* Added a check to prevent ETL from crashing on its first run.

## Steps To Test:
1. Navigate to the home page of the client side application, i.e. http://localhost:3000/attains.
2. Select the _Total Maximum Daily Load_ profile.
3. Confirm that there are 3 accordion sections, and that they operate as expected.
4. Navigate to http://localhost:9090/attains/data/tmdl?columns=completionDate&organizationName=Florida&completionDateLo=06-19-2017&completionDateHi=06-19-2017 and confirm that only rows with `completionDate` equal to `2017-06-19` are returned.
5. Navigate to http://localhost:9090/attains/data/tmdl?columns=completionDate&organizationName=Florida&completionDateLo=06-19-2017 and confirm that no rows with `completionDate` before `2017-06-19` are returned.
6. Navigate to http://localhost:9090/attains/data/tmdl?columns=completionDate&organizationName=Florida&completionDateHi=06-19-2017 and confirm that no rows with `completionDate` after `2017-06-19` are returned.
7. Navigate to http://localhost:9090/attains/data/tmdl?columns=completionDate&organizationName=Florida&completionDateLo=06-19-2017&completionDateLo=06-19-2012. The error `Error! Duplicate completionDateLo parameters not allowed` should be returned.

## TODO:
- [ ] Currently, the duplicate parameter check is being performed only for range inputs. Moving forward, this check will likely be "lifted up", and it should also be performed on `optionsParams`. Additionally, invalid parameters should be checked.

### CODE REVIEW

**Comment by maxdiebold-erg** [comment]
@cschwinderg That last code smell is due to some complex JSX. I'll work on refactoring the input types into separate components, but feel free to start reviewing this whenever you have time.

**Review by cschwinderg — CHANGES_REQUESTED** [review]
Looks good and works well!

I left a comment about simplifying the Accordion component below.

I noticed some other issues (not related to this PR) that I'm going to create some tickets for:
* Alphabetize the options shown in the dropdowns.
* Need to go through and test every filter option on the screen to verify they all work. I went through the Actions and Assessment Units profiles and found the following:
  * Actions
    * Action Agency - Doesn't work at all. The database expects filters to be Tribal, State, or EPA and we are sending T, S, or E. We just need to update the domain values ETL to set the label and value to be the domain name.
    * Parameter - The database has some values that don't show up in the list of possibilities. It looks like the domain values service doesn't have every possibile option. We may need to ask EPA about this, since we may be missunderstanding this domain value. We may also just need to pull from the database and then merge the database results with the domain value.
  * Assessment Units - No issues

File: app/client/src/components/accordion.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  return (
+    <div
+      className={`usa-accordion ${accordionStyles.join(' ')}`}
+      data-allow-multiple={multiSelect}
+    >
+      {children}
+    </div>
+  );
+}
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/components/accordion.tsx:31]: Yep, I'll remove those parameters.

**Comment by maxdiebold-erg** [comment]
@cschwinderg Good point on the input values, I didn't notice the discrepancy with the _Parameter_ values. For the domain services, the field used for the values is currently configured based on the data dictionaries in the "Batch Upload Templates", but I agree, we'll need to test the accuracy of those.

**Review by cschwinderg — APPROVED** [review]
Looks good! Thank you for making those changes.

**Comment by maxdiebold-erg** [comment]
@cschwinderg Good catch, that should be fixed now. Also, I remembered that the EQ team mentioned the "Queries" accordion item (needs to be renamed) should be initially closed, so I pushed that change, too. The accordion item `id` isn't really necessary anymore, but I figured it couldn't hurt.

**Comment by cschwinderg** [comment]
@maxdiebold-erg I'm good with the `id` being there. I'll create a ticket for renaming the "Queries" accordion. Not really sure what we can rename it to. Maybe something like `Advanced API Service URLs`.

**Pull request merged**